src/components/createContainer.jsx

```diff
@@ -0,0 +1,108 @@
+/* @flow */
+
+import React, { Component, PropTypes } from 'react';
+import invariant from 'invariant';
+
+import getDisplayName from '../utils/getDisplayName';
+import createAdaptorShape from '../utils/createAdaptorShape';
+
+const adaptorShape = createAdaptorShape(PropTypes);
+
+export default function createContainer(DecoratedComponent, specs) {
+  const displayName = `AdrenalineContainer(${getDisplayName(DecoratedComponent)})`;
+
+  invariant(
+    specs !== null && specs !== undefined,
+    `${displayName} requires configuration.`
+  );
+  invariant(
+    typeof specs.queries === 'function',
+    `You have to define 'queries' as a function.`
+  );
+
+  return class extends Component {
+    static displayName = displayName
+    static DecoratedComponent = DecoratedComponent
+
+    static contextTypes = {
+      renderLoading: PropTypes.func.isRequired,
+      adaptor: adaptorShape.isRequired,
+    }
+
+    constructor(props, context) {
+      super(props, context);
+
+      const { adaptor } = context;
+      //DecoratedComponent.prototype.shouldComponentUpdate = adaptor.shouldComponentUpdate;
```

Comment:
This is a very important omission. The listener on the store will be re-resolving state after every redux action (which may not have anything to do with Adrenaline). You don't want to update this component and trigger prop changes on the DecoratedComponent unless you really need to.